sdk/aqueduct/backend/api_client.py

```diff
@@ -403,6 +406,14 @@ def preview(
             "dag": dag.json(exclude_none=True),
         }
 
+        if len(body["dag"]) > MAX_REQUEST_BODY_SIZE and any(
```

Comment:
The reason have a check on the size of dag is that there might be multiple `LocalData` in one workflow/preview which causes the http request parsing to error out.